editor/src/clj/util/eduction.clj

```diff
@@ -13,115 +13,130 @@
 ;; specific language governing permissions and limitations under the License.
 
 (ns util.eduction
-  (:refer-clojure :exclude [cat concat dedupe distinct drop drop-while filter interpose keep keep-indexed map map-indexed mapcat partition-all partition-by random-sample remove replace take take-nth take-while]))
+  (:refer-clojure :exclude [cat concat dedupe distinct drop drop-while filter interpose keep keep-indexed map map-indexed mapcat partition-all partition-by random-sample remove replace take take-nth take-while])
+  (:require [util.array :as array]))
 
 (set! *warn-on-reflection* true)
+(set! *unchecked-math* :warn-on-boxed)
 
 (defonce empty-eduction (eduction))
 
 (definline cat [coll]
-  `(eduction
+  `(->Eduction
```

Comment:
`eduction` is a function and creates intermediate `seqs` from the arguments. We can avoid that by constructing the `Eduction` from arguments ourselves when we have exactly two arguments.